server/Mistakes.Journal.Api/Api/Mistakes/Controllers/WeatherController.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Mistakes.Journal.Api.Api.Mistakes.Mappers;
+using Mistakes.Journal.Api.Api.Mistakes.WebModels;
+using Mistakes.Journal.Api.Api.Shared;
+using Newtonsoft.Json.Linq;
+
+namespace Mistakes.Journal.Api.Api.Mistakes.Controllers
+{
+    [ApiController]
+    [Route("/api/weather")]
+    public class WeatherController : ControllerBase
+    {
+        private const string pwRequestPattern = "data/2.5/weather?lat={0}&lon={1}&appid={2}&units=metric";
+        private const string ssRequestPattern = "json?lat={0}&lng={1}&formatted=0";
+
+        private readonly Uri owmApiUri = new Uri("http://api.openweathermap.org");
+        private readonly Uri ssApiUri = new Uri("https://api.sunrise-sunset.org");
+
+        private readonly string owKey = Environment.GetEnvironmentVariable("OPEN_WEATHER_API_KEY");
+
+        [HttpGet]
+        public async Task<ActionResult<WeatherWebModel>> GetWeatherInfo(string lat, string lon)
+        {
+            string weatherJson, sunsetJson;
+
+            try
+            {
+                weatherJson = await GetWeather(lat, lon);
+                sunsetJson = await GetTimeOfDay(lat, lon);
+            }
+            catch (MJException e)
+            {
+                return Problem(type: e.ErrorMessageType.ToString(), detail: e.Message);
+            }
+
+            return Ok(WeatherMapper.ToWeatherWebModel(weatherJson, sunsetJson));
+        }
+
+        public async Task<string> GetWeather(string lat, string lon)
+        {
+            using var client = new HttpClient { BaseAddress = owmApiUri };
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            var request = string.Format(pwRequestPattern, lat, lon, owKey);
+            var response = await client.GetAsync(request);
```

Comment:
It is not obvious to pass such parameters as regex pattern. I would just do it inline to be honest, especially that you nowhere else use the variable.
So:
```suggestion
            var response = await client.GetAsync($"data/2.5/weather?lat={lat}&lon={lon}&appid={owKey}&units=metric");
```

Reply:
Same for time of the day.

Reply:
for me it's obvious, but ok, we can adopt this convention